Packs/Symantec_Advanced_Threat_Protection/Integrations/integration-Symantec_Advanced_Threat_Protection.yml

```diff
@@ -261,14 +275,117 @@ script:
+        if (range_split.length != 2){
+            throw 'date_range must be "number date_range_unit", examples: (2 hours, 4 minutes,6 months, 1 day, etc.)';
+        }
+        var number = parseInt(range_split[0]);
+        var unit = range_split[1].toLowerCase();
+        if (['minute', 'minutes',
+            'hour', 'hours',
+            'day', 'days',
+            'month', 'months',
+            'year', 'years',
+            ].indexOf(unit) < 0) {
+            throw 'The unit of date_range is invalid. Must be minutes, hours, days, months or years.';
+        }
+        var d = new Date();
+        if (unit.indexOf('minute') >= 0) {
+            d.setTime(d.getTime() - (number * 60 * 1000));
+        } else if (unit.indexOf('hour') >= 0) {
+            d.setTime(d.getTime() - (number * 60 * 60 * 1000));
+        } else if (unit.indexOf('day') >= 0) {
+            d.setTime(d.getTime() - (number * 24 * 60 * 60 * 1000));
+        } else if (unit.indexOf('month') >= 0) {
+            d.setTime(d.getTime() - (number * 30 * 24 * 60 * 60 * 1000));
+        } else if (unit.indexOf('month') >= 0) {
+            d.setTime(d.getTime() - (number * 365 * 30 * 24 * 60 * 60 * 1000));
+        }
+        return d;
+    };
 
-    // The command input arg holds the command sent from the user.
+    var getIncidentsFromIncidentsResult = function(res) {
+        return res.map(function(r){
+            return {
+                name: r.summary,
+                rawJSON: JSON.stringify(r)
+            };
+        });
+    };
 
+    var getIncidentsFromEventsResult = function(res) {
+        return res.map(function(r){
+            var name;
+            if ('message' in r){
+                name = r.message;
+            } else if ('uuid' in r){
+                name = r.uuid;
+            } else {
+                name = r.type_id;
+            }
+            return {
+                name: name,
+                rawJSON: JSON.stringify(r)
+            };
+        });
+    };
+
+    var fetchIncidents = function(type, query, limit, first_fetch_time) {
+
+        if (type != 'incidents' && type != 'incidentevents' && type != 'events'){
+            throw 'Unknown incident type.';
+        }
+        var DAY_IN_MILLIS = 24*60*60*1000;
+
+        var lastRun = getLastRun();
+        var now = new Date();
+        var endTime = now.toISOString();
+        var startTime;
+
+        if (lastRun && lastRun.time && lastRun.time !== '') {
+            startTime = lastRun.time;
+        } else if (first_fetch_time) {
+            startTime = parseOffsetDate(first_fetch_time).toISOString();
+        } else {
+            var n = now;
+            n.setTime(now.getTime() - 1 * DAY_IN_MILLIS);
+            startTime = n.toISOString();
+        }
+        var next = false;
+        var p = {verb: 'query'};
+        add(p, 'where', query);
+        addTime(p, 'start_time', startTime);
+        addTime(p, 'end_time', endTime);
+        if (limit) {
```

Comment:
What happens if no limit is inserted as a parameter? This would mean no limit would be set? If so, this could potentially return hundreds of incidents to XSOAR. Please add a default limit of a 200 even if it isn't a parameter.

Reply:
The default limit is defined in `fetch_incidents_limit` configuration (default is 20). I adjusted the style of `limit` according to other functions in the integration.

Reply:
What I mean is if someone removes the `fetch_incidents_limit` completely, the `params.fetch_incidents_limit` will return an empty string (`""`) and will not enter the if statement. If I'm not mistaken this will cause the request to bring all the incidents without any limit. 
![Screen Recording 2020-10-14 at 12 54 24](https://user-images.githubusercontent.com/50324325/95974044-15f2b480-0e1d-11eb-8426-d6632b7ed073.gif)